fe/fe-core/src/main/java/org/apache/doris/catalog/external/EsExternalTable.java

```diff
@@ -0,0 +1,119 @@
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.catalog.external;
+
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.EsTable;
+import org.apache.doris.datasource.EsExternalDataSource;
+import org.apache.doris.external.elasticsearch.EsUtil;
+import org.apache.doris.thrift.TTableDescriptor;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+/**
+ * Elasticsearch external table.
+ */
+public class EsExternalTable extends ExternalTable {
+
+    private static final Logger LOG = LogManager.getLogger(EsExternalTable.class);
+
+    private final EsExternalDataSource ds;
+    private final String dbName;
+    private boolean initialized = false;
+
+    /**
+     * Create elasticsearch external table.
+     *
+     * @param id Table id.
+     * @param name Table name.
+     * @param dbName Database name.
+     * @param ds HMSExternalDataSource.
+     */
+    public EsExternalTable(long id, String name, String dbName, EsExternalDataSource ds) {
+        super(id, name);
+        this.dbName = dbName;
+        this.ds = ds;
+        this.type = TableType.ES_EXTERNAL_TABLE;
+    }
+
+
+    private synchronized void makeSureInitialized() {
+        if (!initialized) {
+            init();
+            initialized = true;
+        }
+    }
+
+    private void init() {
+        fullSchema = EsUtil.genColumnsFromEs(ds.getEsRestClient(), name, null);
+    }
+
+    @Override
+    public List<Column> getFullSchema() {
+        makeSureInitialized();
+        return fullSchema;
+    }
+
+    @Override
+    public List<Column> getBaseSchema() {
+        return getFullSchema();
+    }
+
+    @Override
+    public List<Column> getBaseSchema(boolean full) {
+        return getFullSchema();
+    }
+
+    @Override
+    public Column getColumn(String name) {
+        makeSureInitialized();
+        for (Column column : fullSchema) {
+            if (name.equals(column.getName())) {
+                return column;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String getMysqlType() {
+        return type.name();
+    }
+
+    /**
+     * get database name of hms table.
+     */
+    public String getDbName() {
+        return dbName;
+    }
+
+    @Override
+    public TTableDescriptor toThrift() {
+        return super.toThrift();
+    }
+
+    public EsTable toEsTable() {
+        EsTable esTable = new EsTable(this.id, this.name, this.getFullSchema(), TableType.ES_EXTERNAL_TABLE, name, null,
```

Comment:
I think we don't need to new a `EsTable` everytime we call `toEsTable()`.
You can create it in `init`.